### ISSUE TITLE
Mark an import as failed after some large timeout

### ISSUE THREAD

**Issue by cincodenada** [bookwyrm-social/bookwyrm#1821]
**Is your feature request related to a problem? Please describe.**
I have an import that never actually "finished" - it's at 99% and there's one or two books still pending, but it's been that way for 4 weeks so I doubt it's ever gonna get around do that last book :smile: The key user issue here is that if an import hasn't "finished", I can't re-try imports, which means I can't add books and then recover my reading data (see #1820 for related thoughts there).

**Describe the solution you'd like**
After some timeout, I think we should mark any pending imports as failed. Ideally this wouldn't happen in the first place, but there's always the chance of something going wonky, and this would be a good backstop.

**Describe alternatives you've considered**
Alternately, we could give the user the option to "give up" on an import, and mark any pending books as failed.

**Comment by mouse-reeve** [comment]
Agreed -- it's weirdly hard to tell when an import has completed because of how the tasks are distributed. There is logic that should let you retry an individual item if the job is has no movement for 24 hours. Have you seen that show up?
